pkg/module_manager/module_hook.go

```diff
@@ -165,6 +166,19 @@ func (h *ModuleHook) Run(bindingType BindingType, context []BindingContext, logL
 		return err
 	}
 
+	if len(hookResult.KubernetesPatchBytes) > 0 {
+		var specs []object_patch.OperationSpec
+		specs, err = h.moduleManager.KubeObjectPatcher.ParseSpecs(hookResult.KubernetesPatchBytes)
+		if err != nil {
+			return err
+		}
+
+		err = h.moduleManager.KubeObjectPatcher.GenerateFromJSONAndExecuteOperations(specs)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Can it be just one method, as a `SendBatch` for metrics?

Reply:
Have I made the same mistake in the shell-operator PR?

Reply:
Yes, the same logic applies to a `Hook.Run` method in the shell-operator.

https://github.com/flant/shell-operator/pull/239/files#diff-d6e8286d4f35ebf5243c1d71976c0114b0d7db3bf7dca16dbfb5f034b351ec4cR137-R156

Reply:
I believe, there is a simpler solution. Also pushed it to the shell-operator PR.
https://github.com/flant/addon-operator/pull/177/commits/c8b67e5533f01705cce6cc243145b83c07e5cc59

Reply:
Yeah, it is better now.

The problem that I still see here is that the `KubernetesPatchBytes` in hookResult affect go-hooks in a way that we need to generate and return patches in `[]byte` form instead of handy struct [OperationSpec](https://github.com/flant/shell-operator/pull/239/files#diff-726461c73ad5fd14bc1f28d156c4fc31e5675c340c767552491edd9b40f73fd4R3-R15). See https://github.com/flant/addon-operator/blob/master/sdk/sdk.go#L53-L58. _As I know, there is a new sdk, so maybe bytes in HookResult are not an issue._

Reply:
Ah, that makes sense. Fixed here and in the shell-operator!